spatial_maps/maps.py

```diff
@@ -78,11 +81,19 @@ def occupancy_map(self, smoothing):
         return smooth_map(self.time_pos, self.bin_size, smoothing)
 
     def rate_map(self, smoothing, mask_zero_occupancy=False):
-        spike_map = self.spike_map(smoothing)
-        occupancy_map = self.occupancy_map(smoothing)
+        spike_map = self.spike_map(smoothing=0)
+        occupancy_map = self.occupancy_map(smoothing=0)
```

Comment:
@lepmik I think it's better to compute spike map and oc map without smoothing, and then smooth the rate map.

Reply:
No, then you will get edge effects - which cancel out (not perfectly of course) when you smooth both maps and then divide.

Reply:
So here's the problem: if I use `mask_zero_occupancy=True` I correctly get this:

`rate_map = smap.rate_map(smoothing=0.04, mask_zero_occupancy=True)`
![image](https://user-images.githubusercontent.com/17097257/64948442-5f1db100-d877-11e9-977b-4af7b360f97f.png)

When I use `mask_zero_occupancy=False` I get (with smoothing of spike map and occ map):

`rate_map = smap.rate_map(smoothing=0.04, mask_zero_occupancy=False)`
![image](https://user-images.githubusercontent.com/17097257/64948512-88d6d800-d877-11e9-8fe3-9306e5d1090e.png)

If I compute spike and occupancy maps with no smoothing and smooth the rate map I get this:
![image](https://user-images.githubusercontent.com/17097257/64948651-c5a2cf00-d877-11e9-9bfb-d2014014b02e.png)

which I think it's better! Do you agree?

I think that the edge effects would not really cancel out as you divide bin by bin anyways

Reply:
Ok I see what you mean with the edge effect.. true you get an edge effect if you smooth afterwards. But still he second plot doesn't look good. 

What about: smoothing before AND setting the unoccupied bins of the non-smoothed occupancy map to zero?

Reply:
Yes, that is totally possible, but I do not think that is to be done by default. In your case you have lots of "missing" data. There are many ways to deal with that, e.g. the missing points inside your field would be best to interpolate. As you have access to the occupancy map from the SpatialMap class I suggest this is best dealt with by the user. If you for example make a mask which covers the inside of the track, then the rate on the track as obtained in the second plot is imo a better approximation than if you set zero occupancy to zero.

Reply:
ok! what about adding a `mask` argument? the atropy convolution has the same arg and it could be used when smoothing to avoid this behavior.

So the rate_map funciton will be unchanged unless a `mask` is passed.

Reply:
or maybe having a `convolve_kwargs` or `**kwargs` which is sent directly to astropy convolution?